src/Common/src/Interop/OSX/System.Native/Interop.ProtocolStatistics.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [StructLayoutAttribute(LayoutKind.Sequential)]
+        public unsafe struct TcpGlobalStatistics
+        {
+            public readonly ulong ConnectionsAccepted;
+            public readonly ulong ConnectionsInitiated;
+            public readonly ulong CumulativeConnections;
+            public readonly ulong ErrorsReceived;
+            public readonly ulong FailedConnectionAttempts;
+            public readonly ulong SegmentsReceived;
+            public readonly ulong SegmentsResent;
+            public readonly ulong SegmentsSent;
+            public readonly int CurrentConnections;
+            private readonly int __padding;
+        }
+
+        [DllImport(Libraries.SystemNative)]
+        public static unsafe extern int GetTcpGlobalStatistics(out TcpGlobalStatistics statistics);
+
+        [StructLayoutAttribute(LayoutKind.Sequential)]
+        public unsafe struct IPv4GlobalStatistics
+        {
+            public readonly ulong OutboundPackets;
+            public readonly ulong OutputPacketsNoRoute;
+            public readonly ulong CantFrags;
+            public readonly ulong DatagramsFragmented;
+            public readonly ulong PacketsReassembled;
+            public readonly ulong TotalPacketsReceived;
+            public readonly ulong PacketsDelivered;
+            public readonly ulong PacketsDiscarded;
+            public readonly ulong PacketsForwarded;
+            public readonly ulong BadAddress;
+            public readonly ulong BadHeader;
+            public readonly ulong UnknownProtos;
+            public readonly int DefaultTtl;
+            public readonly int Forwarding;
+        }
+
+        [DllImport(Libraries.SystemNative)]
+        public static unsafe extern int GetIPv4GlobalStatistics(out IPv4GlobalStatistics statistics);
+
+        [StructLayoutAttribute(LayoutKind.Sequential)]
+        public unsafe struct IPv6GlobalStatistics
+        {
+
```

Comment:
Yeah this is an unused type, I will remove. I thought that OSX reported different statistics for IPv4 and IPv6, but it turns out that the structure returned from the net.inet6.ip6.stats sysctl call isn't actually contained in any header files on the OS. I found some document stating that it was a private structure and that reading from it wasn't supported.